services/explorer-indexer/src/types/rpcWs/RpcWsResponse.utils.ts

```diff
@@ -0,0 +1,110 @@
+import { Logger } from '@nestjs/common';
+import { WebSocketClient } from 'nestjs-websocket';
+import util from 'util';
+import { ResponseTypes, Topics } from './RpcWsResponse.enums';
+import {
+  BaseResponseRype,
+  NewDagBlockFinalizedResponse,
+  NewDagBlockResponse,
+  NewPbftBlockHeaderResponse,
+  NewPbftBlockResponse,
+  RpcResponseData,
+} from './RpcWsResponse.interfaces';
+
+export function checkType(object: BaseResponseRype): string {
+  if (!object) return;
+  return object.subscription
+    ? object.subscription === '0x1'
+      ? ResponseTypes.NewDagBlockResponse
+      : object.subscription === '0x2'
+      ? ResponseTypes.NewDagBlockFinalizedResponse
+      : object.subscription === '0x3'
+      ? ResponseTypes.NewPbftBlockResponse
+      : object.subscription === '0x4'
+      ? ResponseTypes.NewHeadsReponse
+      : object.subscription === '0x5'
+      ? ResponseTypes.NewPendingTransactions
+      : ''
+    : '';
+}
+
+export const toObject = (
+  data: WebSocketClient.Data,
+  unrecognizedDataCb: (msg: string) => void,
+  logger: Logger
+) => {
+  const response = JSON.parse(data.toString()) as RpcResponseData;
+
+  switch (response.result) {
+    case '0x1':
```

Comment:
Same here. `Subscriptions` enum.

Reply:
Fixed in https://github.com/Taraxa-project/taraxa-platform/pull/277/commits/5377fde50d42713cd52037847e62fedfdb430f48